### ISSUE TITLE
how to center or align the image in image-gallery-image

### ISSUE THREAD

**Issue by jacobeverist** [xiaolin/react-image-gallery#159]
I overrode the default CSS with some custom CSS to limit the maximum size of the image.  Previously it was expanding to the entire container which caused portrait images to drop below browser window.

I am trying embed a custom video application in the image gallery, but I am struggling with centering the thumbnail and the embedded player.  Can you give advice?

Here is my custom CSS:
```
.kaltura-video-slide .image-gallery-slide img {
  width: auto;
  height: 330px;
  margin: 0 auto;
}

```

I created a custom renderItem function for this:
```
  _renderKaltura(item) {
    return (
      <div className='image-gallery-image'>
          <KalturaVideo firstName={item.firstName} targetId={item.targetId} partnerId={item.partnerId} uiconfId={item.uiconfId} entryId={item.entryId}/>
              {item.firstName}
        {
              item.description &&
        <span
          className='image-gallery-description'
          style={{right: '0', left: 'initial'}}
        >
          {item.description}
        </span>
        }
      </div>
    );
  }
```

Finally, the result is shown here.  I've tried monkeying with the CSS on all of the classes, but I can't figure out what will make the player or image to center.  I have the same behavior without this custom rendering using only images.

<img width="609" alt="screen shot 2017-03-05 at 10 03 06 pm" src="https://cloud.githubusercontent.com/assets/5069498/23598589/52c2513c-01f0-11e7-987c-7cc820128ed3.png">

**Comment by xiaolin** [comment]
try adding 

```
.image-gallery-slides {
  text-align: center;
}
```

**Comment by jacobeverist** [comment]
Sadly that didn't work.

One thing I noticed is that the image is on the left and the description is all the way to the right.   Seems like there is some weird interaction going on.

I'm looking at this section in your CSS, and trying to figure what might be happening.  Perhaps it is reacting with other CSS in my application.

```
.image-gallery-slides {
  line-height: 0;
  overflow: hidden;
  position: relative;
  white-space: nowrap; }

.image-gallery-slide {
  background: #fff;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%; }
  .image-gallery-slide.center {
    position: relative; }
  .image-gallery-slide img {
    width: 100%; }
  .image-gallery-slide .image-gallery-description {
    background: rgba(0, 0, 0, 0.4);
    bottom: 70px;
    color: #fff;
    left: 0;
    line-height: 1;
    padding: 10px 20px;
    position: absolute;
    white-space: normal; }
    @media (max-width: 768px) {
      .image-gallery-slide .image-gallery-description {
        bottom: 45px;
        font-size: .8em;
        padding: 8px 15px; } }
```

**Comment by xiaolin** [comment]
Just double checking that you added text-align property to `image-gallery-slides` and not `image-gallery-slide` those two class names are very similar.

**Comment by jacobeverist** [comment]
I did correct.  Just to be sure, I did it again.  I added the field like so with no change in the rendering result:

```
.image-gallery-slides {
  line-height: 0;
  overflow: hidden;
  position: relative;
  white-space: nowrap;
  text-align: center;
}
```

**Comment by xiaolin** [comment]
Thats odd, I tried the same using the demo url and adjusting the width on the images to be smaller and it worked. Maybe some issues with css not being applied or being overwritten. Can you inspect the element and check the styles were applied?

**Comment by jacobeverist** [comment]
When I reverted to the demo code, the images centered just fine with the `text-align: center` tag.  So I know there was something strange going on with the embedded video.

Figured it out.  2 things I had to solve.  

First, the <div> containing the iframe of the embedded video was using the Bootstrap "row" class which was overriding the margins and fixing it to the left side of the screen.

Second, for the <div> containing the iframe, I added the following style which auto-centered the video:

```
.kWidgetIframeContainer{
  margin: 0 auto;
}
```

Where the class name is for the Kaltura video javascript player library.

Thanks for the help!
